lib/rspec/mocks/verifying_proxy.rb

```diff
@@ -34,23 +57,25 @@ def add_message_expectation(method_name, opts={}, &block)
         ensure_implemented(method_name)
         super
       end
+    end
 
-      def ensure_implemented(method_name)
-        return unless method_reference[method_name].unimplemented?
+    # @private
+    module MaybeVerifyProxyMethods
```

Comment:
Is it really necessary to introduce all these `Maybe...` and `Could...` modules and classes?  They make the ancestry of these classes really confusing and it's not clear to me why they are needed.  On top of that, given that you still check `source_space.temporarily_suppressing_verification?`it seems like these classes only have different behavior if `temporarily_suppressing_verification` is set when a stub or message expectation is created--but in that case, why do we need different classes at all?  For example, I think users would rightly expect this to work:

``` ruby
dbl = double("MyClass")
without_verifying_partial_doubles do
  allow(dbl).to receive(:some_method_that_would_fail_verification)
end
```

...but as implemented, it looks to me like that won't work.

I think it would be better and simpler to just _always_ check `temporarily_suppressing_verification` before calling `ensure_implemented` (or maybe check it in `ensure_implemented` itself so we don't have to duplicate the check in the callers).  Then it seems like there'd be no need for these extra confusing modules and classes.